core/chains/evm/client/rpc_client.go

```diff
@@ -412,8 +435,46 @@ func (r *rpcClient) BatchCallContext(ctx context.Context, b []rpc.BatchElem) err
 	duration := time.Since(start)
 
 	r.logResult(lggr, err, duration, r.getRPCDomain(), "BatchCallContext")
+	if err != nil {
+		return err
+	}
 
-	return err
+	// populated requested finalized block with correct value
```

Comment:
Maybe mention Astar here... or alternatively move comment inside the if statement that follows?

Also, populated -> populate